### ISSUE TITLE
Ways to involve collaborators in CTC meetings

### ISSUE THREAD

**Issue by rvagg** [nodejs/CTC#10]
I'd like to talk about how we might make our meetings a bit more inclusive to @nodejs/collaborators to help blur the line a little between the CTC and everyone else and reduce the perception of it being an elite group and more about it simply being a way to make difficult decisions.

To date, our problems have been around the tools available to host meetings. An ideal would be for meetings to be open to join by _anyone_ but we'd need a way to moderate discussion to ensure that the CTC could actually get stuff done (i.e. observers would mostly have to observe). We've yet to find a tool that is reasonably priced for this purpose but we keep on exploring. Right now we use Uberconference and have a line that we use each week to host the meetings and we keep the connection details private so we don't get swamped by people that we can't properly moderate.

So, here's some suggestions for possible changes to how we do meetings:
- Find a better technology - we can ask the Foundation to pay for a solution if we find something that's not too costly, what we've reviewed so far either hasn't suited our needs or has been too expensive.
- Open meetings to all collaborators to join whenever they want as observers
- Have a procedure for allowing collaborators to request to join a meeting beforehand so we can do some gating to try and keep things productive. Perhaps they have to send a private request to the CTC to ask to join some period of time before the meeting.

On the last point, one concern of making them completely open is that our collaborator list is quite long, so it's very possible that meetings could be made unproductive by having too many voices participating. I can imagine this happening whenever hot topics are discussed, even with people misusing their collaborator status to get a disproportionate voice (remember that the collaborator list includes people who have effectively moved on, perhaps that's another discussion). Our meetings usually push the 1h mark and sometimes beyond, it's not going to work if we make that any worse.

**Comment by Trott** [comment]
TL;DR: Read "non-voting participant" where you see "observer".

Some points to clarify some things that not everyone may be aware of (although I'm sure all CTC members are aware):
- We (most of the time) manage to make it possible for _anyone_ (collaborator or not) to be observers by watching the YouTube live stream.
- So what we're talking about here is making it possible for people to be participant-observers, even though we just call them "observers" for shorthand. A good example is @bmeck and @jdalton in the last several meetings as "observers" to talk about the Unambiguous JavaScript proposal and ES modules. They were really more "non-voting participants" than "observers". They answered a lot of questions, and framed the discussion. So, don't get confused by the "observer" label. We're talking about involvement. The non-members just don't get a vote.
- We (or at least I) would expect that most "non-voting participant-observers" would be involved in a single topic on the typical meeting agenda and perhaps not the whole meeting. I'm not sure there's a reasonable way to alter the way meetings are run and technology is used to take advantage of this fact to make things better for everyone, but putting it out there in case someone else has a great idea.

**Comment by williamkapke** [comment]
**EDIT**: I just revisited the README and saw the request for these topics to be limited to members of the CTC and the Collaborators Group (which I am not). I apologize for not remembering / realizing beforehand. Feel free to remove with no hard feelings.

---

Interesting this conversation cropped up. I mentioned to @nebrius last night about how "Observer" is used all the time but not defined anywhere- so I'm extremely happy to see this!

My thoughts:
- This seems like it could apply to many groups in the org. @rvagg Are there things that are unique to the CTC on this topic?
- Strive to find a list of action items that lead to documenting the outcome here as policy. (Don't let the be hidden in this issue please!)
- "non-voting participants" (Observers) have an important distinction from the YouTube observers: **They attend the private session.**

> Find a better technology - we can ask the Foundation to pay for a solution if we find something that's not too costly, what we've reviewed so far either hasn't suited our needs or has been too expensive.

@rvagg How much are we talking here? Consider soliciting donations / sponsorship too? Would the use of this new solution be unique to the CTC? Hm... I guess these questions go too deep here (derail) and are better served on the https://github.com/nodejs/TSC/issues/48 thread. Consider commenting there?

> We (or at least I) would expect that most "non-voting participant-observers" would be involved in a single topic on the typical meeting agenda and perhaps not the whole meeting.

I agree that this makes sense - except that I'm <del>concerned</del> curious about the use of "**most** [..observers]".<br>@Trott Just wondering, who would be exempt from this & why?<br>If this was made official, I hope a standard (polite) way is established to handle someone speaking out of turn to avoid others feeling like the person is getting special treatment. 

> Perhaps they have to send a private request to the CTC to ask to join some period of time before the meeting.

This is really tough since rejection or being ignored is a killer. This props up "the elite group" view unfortunately.

> Our meetings usually push the 1h mark and sometimes beyond, it's not going to work if we make that any worse.

To mitigate this, some organizations emphasize that meetings are for clarification and voting rather than brainstorming which can be done in other (possibly better) venues. With that, "non-voting participant-observers" serve as a resource if clarification is needed by a voting member.

**Ultimately**, I don't see this being feasible without making the meetings more formal... which may not be too favorable.

**Comment by Trott** [comment]
> @Trott Just wondering, who would be exempt from this & why?
> If this was made official, I hope a standard (polite) way is established to handle someone speaking out of turn to avoid others feeling like the person is getting special treatment.

We have some standing observers who are always invited who would typically be exempt. An example is the observer from the V8 team. One reason they are there is to surface and resolve instances where Node.js and V8 may be working at cross-purposes without even realizing it.

My point, though, was _not_ that people should be silenced for speaking during the wrong issue or something like that. It was more like: "Hey, chances are any observers (aside from the ones that we currently have invited to every meeting) will only be interested in a part of the meeting. So if there's an organizational and technological way to include them for just the 15 minutes they care about rather than having them sit around and listen to a half hour of conversation on other topics first, then that's a win for them." An alternative perspective, of course, is that the more people hear about _all_ topics, the better, so I guess there's two sides to that. 

One obvious idea there might be to put times on the agenda. Sometimes it's easy to predict how long a particular topic will take, but not always, and it's not always wise to cut off conversation when things are going, so I'm not sure that will be deemed workable by the CTC. I certainly have my reservations, but would be willing to give it a try if it was helpful to people. But at this point, it's probably putting the cart in front of the horse.

**Comment by evanlucas** [comment]
@williamkapke 

> "non-voting participants" (Observers) have an important distinction from the YouTube observers: They attend the private session.

That is not always true. There have been cases where "observers" were asked to drop off (security related issues, etc). I definitely do agree that they are still different since it is not a 2-way conversation like being in the actual conference call/hangout though.

**Comment by mikeal** [comment]
> Find a better technology - we can ask the Foundation to pay for a solution if we find something that's not too costly, what we've reviewed so far either hasn't suited our needs or has been too expensive.

What are the problems we have with youtube broadcasting? I'd like to enumerate them so that we have a list of things we are looking for out of alternatives.

**Comment by trevnorris** [comment]
With this in mind, can we also do something about the stand-up? IIRC we've been asking observers to participate in the stand-up as well, and even w/o their participation it's taking a good chunk of the meeting. TBH I feel like we need to be more disciplined about what we share. If it takes more than two sentences then reconsider whether it's really worth sharing. Or maybe we just drop that section completely and have participants simply add it to the meeting notes.

**Comment by mikeal** [comment]
> With this in mind, can we also do something about the stand-up?

Ya, this has gotten a little unsustainable. Perhaps we should identify areas of the code base or working groups that we want updates on and assign people to do a standup about them rather than get something from every person.

**Comment by trevnorris** [comment]
@evanlucas 

> I definitely do agree that they are still different since it is not a 2-way conversation like being in the actual conference call/hangout though.

Let's break down the groups of observers to: "casual observers", those who want to simply see what's going on, "concerned observers", those who have an opinion on a decision but aren't actually involved, and "participating observers", those we call in because they are directly involved in an issue the CTC will need to vote on (modules), or that node depends on (v8). As @Trott mentioned, this varies based on participant and a given topic.

It's not sustainable to allow anyone who wants to share a comment to share a comment. If a "concerned observer" comes with strong opinions they might not want to give up the floor on a topic. Should we allow them to disrupt the vote, or should we ask them to either hold their opinions and/or leave so the vote can take place?

This is one hypothetical example of having potential issues, but also trying to stay within the realm of possibility. Now, not saying they should stay excluded, but am saying that simply allowing anyone who wants to speak up to speak up could be a problem.

I feel like key participation has to do with when the CTC needs to vote. For this it may be reasonable to give a first week, "hey everyone, we'll be voting on topic X next week, here's the issue number where it can be discussed". Allow that to get hashed out and the following week we can recap whether we feel the thread is to a point where voting makes sense. We sort of do that already, but formalizing it may help.

**Comment by cjihrig** [comment]
> Or maybe we just drop that section completely and have participants simply add it to the meeting notes.

+1

**Comment by eljefedelrodeodeljefe** [comment]
At this point I advise seeking professional help of some agile coaches or so. Listening to your meetings, I am pretty sure a formalized and / or professional meeting conduct would help tremendously. Without being blameful, but the meetings often leave the impression that nobody has prepared them or themselves decently and jumping off the appointment 60min prior to meetings supports that even more, given that there are so many people being payed for this. Also there are lots of pauses in the talking which is hard to follow and adds lengths. Naturally long meetings lead to heated to discussions towards the end - humans.

Frankly I would advise against opening this up before that hasn't been addressed. Also some private or at least unrisky environment for your discussion is a good thing. 

I fear that opening up this meeting won't do anything against the elitarian impression. My two cents.

**Comment by Fishrock123** [comment]
> the meetings often leave the impression that nobody has prepared them

Most of us already occupy our time trying to get stuff done.

> given that there are so many people being payed for this

Paid to work on core productively, not paid to make meetings, uh, "feel smooth".

Also, If I count correctly, I'm pretty sure just under half of the CTC isn't (directly at least).

> Also there are lots of pauses in the talking which is hard to follow and adds lengths.

_wow I wonder if some people aren't introverts_

This allows space for us to make sure everyone's voice is heard, thanks.

## 

We usually get everything sorted out within the hour. Perhaps It could be smoothened out a little bit, but this doesn't really function like your run-of-the-mill meeting either.

**Comment by jasnell** [comment]
I hardly think that snarky comments here are going to help the conversation much. @rvagg opened the conversation here to solicit input from the collaborator base and that's what @eljefedelrodeodeljefe is providing. Reacting to his feedback with sarcasm doesn't move the conversation along productively.

Exceedingly few meetings are going to go as smoothly as one would like and we all are all quite busy balancing several responsibilities... paid or otherwise. The process can always get better and it serves our interests to incrementally improve the process over time. 

Ditching the stand-up would be worthwhile but I would still like the stand-up to be included in the minutes. I like the trend we've had recently of people dropping their stand-up comments into the minutes on their own. If we can keep that up, meetings would likely go far smoother.

**Comment by Trott** [comment]
@williamkapke:

> EDIT: I just revisited the README and saw the request for these topics to be limited to members of the CTC and the Collaborators Group (which I am not). I apologize for not remembering / realizing beforehand. Feel free to remove with no hard feelings.

I think your comments and feedback are welcome. That stuff in the README says it's _primarily_ for collaborators but not _exclusively_. My opinion only, but I see it as mostly to give license to locking issues when they become unproductively noisy. This one has (mostly) not reached that phase yet. Speaking only for myself, I would welcome your continued participation in this conversation. You usually have either good ideas or serve to highlight where we are not communicating effectively or making assumptions. You often manage to do _all_ of those things.

**Comment by bnoordhuis** [comment]
> Listening to your meetings, I am pretty sure a formalized and / or professional meeting conduct would help tremendously.

You make it sound like a CTC meeting is a performance in front of a live audience.  It's not, it's (for the most part) an informal get-together for synchronous discussion.  That people can listen in is nice but secondary.

> Also there are lots of pauses in the talking which is hard to follow and adds lengths.

That's teleconferencing for you: no ambient clues that tell you if now is a good time to start or stop speaking.

**Comment by eljefedelrodeodeljefe** [comment]
@Fishrock123 I am going to ignore this. But feel to read my reply to @bnoordhuis comment that takes up some of the points.

@bnoordhuis re performance and audience. Of course it's not. But there is justification for professional and trained managers to steer engineering teams. One tool of theirs is a productive meeting that has good flow, is short and yields results. I guess every tech person would agree that those are goals of a good tech meeting too and this can be applied to CTC meetings as well, even though it is informal. Having heard much of a year of those meetings, often it seems that people get heated out of irrational reasons such as boredom or other frustration not connected to the matter at hand.

Re technology or ambience. This is why advanced tele working companies are spending lots of budgets on modern video call technology and / or personnel that is trained to conduct meetings such as this. E.g. and outside moderator, that is not affected by the contents of the meetings but quickly gets a clue who would need to start talking to progress the discussion.

All in all, it's rather about human "deficiencies" rather than organizational, as proposed and discussed. You guys are ordinary humans as well and could take what modern management offers, instead of having none. This could then be as beneficial and increasing happiness to you as it does to the org.

**Comment by bnoordhuis** [comment]
> modern video call technology

The implicit assumption in that statement is that everyone has a fast internet connection.

My connection currently tops out at 0.5 MB/sec, and only when the wind is just right.  People frequently dial in over hotel wifi that is even worse.

**Comment by eljefedelrodeodeljefe** [comment]
That statement was "and / or"-connected with the moderator.

**Comment by rvagg** [comment]
@williamkapke re cost, I wouldn't index too much on that barrier tbh, we can get the Foundation to pay for something if we find a perfect solution that solves all our needs. There are some that are unreasonable, however, like BlueJeans, mainly because we want to be able to broadcast and those solutions are priced as if we are doing webinars or training, which we are not.

I'm currently poking at http://zoom.us/ which offers a decent product, they do have a bit of a problem with broadcast pricing for the same reason as BlueJeans but at least it's not insane pricing.

Also, we _could_ drop the need for broadcasting and use the solution we have now, but it's pretty hacky and manual and requires one of the individuals on the call to have the software running—I believe that it works best when running on a separate computer to the one you're using, so that's not great.

Ideally, any solution we adopt could serve the whole org, or at least the most active parts of it, so the different working groups can make use of it if they want.

Also, to be absolutely clear about the scope of this discussion—I was proposing allowing collaborators to join in an interactive but non-voting (thanks Rich). Not just being able to watch via live-stream. Not including others outside of the collaborator group. So I'm not suggesting a free-for-all, just widening the tent a little.

**Comment by Trott** [comment]
Closing. Feel free to re-open or comment if you think this conversation should continue.
